File: module_java/src/main/java/org/gmfbilu/superapp/module_java/basic/单例/Singleton2.java
package org.gmfbilu.superapp.module_java.basic.单例;

/**
 * 懒汉式线程安全,效率低.单例一直没用过，但是这个东西还是一直占用我的内存
 */
public class Singleton2 {

    private static Singleton2 sSingleton2;

    private Singleton2() {
    }

    public static synchronized Singleton2 getSingleton2() {
        if (sSingleton2 == null) {
            sSingleton2 = new Singleton2();
        }
        return sSingleton2;
    }
}

File: module_kotlin/src/main/java/org/gmfbilu/superapp/module_kotlin/kotlin/helloWorld/设计模式/Singleton2.kt
package org.gmfbilu.superapp.module_kotlin.kotlin.helloWorld.设计模式

/**
 * 懒汉式线程安全,效率低.单例一直没用过，但是这个东西还是一直占用我的内存
 */
class Singleton2 private constructor() {

    companion object {
        private var instance: Singleton2? = null

        @Synchronized
        fun get(): Singleton2 {
            if (instance == null) {
                instance = Singleton2()
            }
            return instance!!
        }
    }
}